### PR TITLE
RevEng: Remove redundant warnings & checks

### LINKED ISSUE #9293: ToListAsync never returns back after query execution
We just upgraded to EF Core 2.0 Preview 2 and I have seen this issue in EF COre 1.1 

For some of our queries we want to fetch parent records, but then we also find the related child records in one db call. 

so we have our query outlined as this

await query.Select(parent=> new 
{
ParentPropertiesProjections..
ChildCollectionModel = parent.ChildCollection.Select(t=> ....)
}.ToListAsync();

I see in the logs that EF does fire the query to get all the parent tables data, but doesn't produce any more logs after that and then the query never returns back. 

If this scenario is not supported, then the EF core framework should throw an exception for such instead of silently killing the thread. 

I don't see anything in logs at all to provide you details on it. 

Any guidance is greatly appreciated. 

I am running EF Core 2.0 on Windows 10 machine with VS 2017 Enterprise 15.3 Preview 6 against Sql Server

### LINKED ISSUE #9323: RevEng: Default/Computed Value processing
At present, DefaultValueSql does not set ValueGenerated on DatabaseColumn but ComputedValueSql sets it to AddOrUpdate.
If we are assuming that calling `HasDefaultValueSql` will set ValueGenerated by itself then same should also be true for `HasComputedValueSql` hence setting ValueGenerated for it is redundant. 
If it is incorrect assumption then, DefaultValue should set to ValueGenerated.OnAdd.

Identity must set ValueGenerated.OnAdd & rowversion must set OnAddOrUpdate.

### LINKED ISSUE #9333: RevEng: Warnings for null names for objects which requires name
We have "warnings" for sequence/column/index not having names but it is impossible to create them without name. There seem to be no way to actually hit that code path. Should we have warnings for them?

### LINKED ISSUE #9334: RevEng: Warnings for skipped table
At present when we encounter index/fk from a table which is not included by user, we generate "warning" that we couldn't find the table. That means for a skipped table we would generate multiple warnings even though that was user intent. Should we remove this unnecessary warnings?

### PR DESCRIPTION
**Pull request by smitpatel**
Set metadata appropriately

Resolves #9323 
resolves #9333 
resolves #9334

### CODE REVIEW

File: src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs [hunk 1]
```diff
@@ -294,12 +294,6 @@ private void GetSequences()
                         DisplayName(sequence.Schema, sequence.Name), sequence.StoreType, sequence.IsCyclic,
                         sequence.IncrementBy, sequence.StartValue, sequence.MinValue, sequence.MaxValue);
 
-                    if (string.IsNullOrEmpty(sequence.Name))
-                    {
-                        Logger.SequenceNotNamedWarning();
-                        continue;
-                    }
-
```
Comment (bricelam) [inline src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs]: :tada:

File: src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs [hunk 2]
```diff
@@ -458,24 +451,17 @@ ELSE null
...
 
                     if (!_tables.TryGetValue(SchemaQualifiedKey(tableName, schemaName), out var table))
                     {
                         Logger.ColumnSkipped(DisplayName(schemaName, tableName), columnName);
                         continue;
                     }
 
-                    if (string.IsNullOrEmpty(columnName))
-                    {
-                        Logger.ColumnNotNamedWarning(DisplayName(schemaName, tableName));
```
Comment (bricelam) [inline src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs]: Can we remove these resources and APIs? If not, we should mark them as obsolete.

Comment (smitpatel) [inline src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs]: I cannot remove the APIs ( 😢 )
The logger methods are in internal namespace. The resources are in internal namespace. but there is a test which complains that for each event id there should be logger method with same name. Are we supposed to mark things in internal namespace as obsolete? 
EventIds are public so cannot break them.

Comment (bricelam) [inline src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs]: Maybe we should allow marking event IDs as obsolete in the test so it doesn't fail for them.

Comment (smitpatel) [inline src/EFCore.SqlServer/Scaffolding/Internal/SqlServerDatabaseModelFactory.cs]: I am leaving this as of for now. Will file an issue.

**Pull request merged**